ground/src/test/java/com/google/android/ground/model/job/TaskNotFoundExceptionTest.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.model.job
+
+import com.google.android.ground.BaseHiltTest
+import com.google.common.truth.Truth.assertThat
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner::class)
+class TaskNotFoundExceptionTest : BaseHiltTest() {
+
+  @Test
+  fun testCustomLocalizedMessage() {
+    val obj = Job.TaskNotFoundException(taskId = "1")
+    assertThat(obj.localizedMessage).isEqualTo("unknown task 1")
+  }
+}
```

Comment:
I don't feel we need a specific test class for the exception. Instead, we should look into how to add coverage for this class using user flows.

Reply:
ok, this class was mentioned in the given test coverage list.


> how to add coverage for this class using user flows.

I think an independent is useful to confirm whether the custom message is correct or not. No matter which user flow is uses.

Reply:
The custom message is not user facing. @sufyanAbbasi Wdyt?

Reply:
My apologies on this, I did include this on the list because it showed up in the test coverage class list but it does look like something we can probably skip, since the code under test is a one-liner:

```
class TaskNotFoundException(taskId: String) : Throwable(message = "unknown task $taskId")
```

I will look into making sure that it can be removed from the test coverage class list. If I can do that, then we can go ahead and remove this one from the list. If that's not possible, we can keep the test in to satisfy the test coverage runner.

Reply:
ok, IMO it's ok to have a test, no matter class is a one liner, just to save time in future and if the custom message is changed accidently, the test file will be our reminder. 

Will close the PR as we are not planning to have it.